### PR TITLE
Blindly adds partial cross-language intelligibility

### PR DESCRIPTION
**Pull request by Anewbe**
Bam, random blind port of something that's not even merged on its actual GitHub:
https://github.com/Baystation12/Baystation12/pull/24192

~~I have wanted this for *ages*.~~

### CODE REVIEW

**Comment by elgeonmb** [comment]
i will probably be tweaking some of these percent chances, should i do that here or just do a seperate PR afterwards

**Comment by Anewbe** [comment]
Later.

**Comment by elgeonmb** [comment]
figured as much

**Comment by HeyBanditoz** [comment]
Slow down

File: code/modules/mob/language/language.dm [hunk 1]
```diff
@@ -41,8 +42,38 @@
...
+			understand_chance += L.partial_understanding[name] * 0.5
+	var/scrambled_text = ""
+	var/list/words = splittext(input, " ")
+	for(var/w in words)
+		if(prob(understand_chance))
+			scrambled_text += " [w] "
+		else
+			var/nword = scramble_word(w)
+			var/ending = copytext(scrambled_text, length(scrambled_text)-1)
+			if(findtext(ending,"."))
```
Comment (Atermonera) [inline code/modules/mob/language/language.dm]: Should this check for other full-stops, like "?" and "!" ?

Comment (Anewbe) [reply code/modules/mob/language/language.dm]: It probably should, yes.

**Pull request merged**